src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTranslationAction.Actions.cs

```diff
@@ -31,15 +31,7 @@ public override Task<Compilation> InvokeAsync(Compilation oldCompilation, Cancel
                 public DocumentId DocumentId => _newState.Attributes.Id;
             }
 
-            private class RemoveAllDocumentsAction : CompilationTranslationAction
-            {
-                public override Task<Compilation> InvokeAsync(Compilation oldCompilation, CancellationToken cancellationToken)
-                {
-                    return Task.FromResult(oldCompilation.RemoveAllReferences());
-                }
-            }
-
-            private class RemoveDocumentAction : SimpleCompilationTranslationAction<DocumentState>
+            internal class RemoveDocumentAction : SimpleCompilationTranslationAction<DocumentState>
```

Comment:
why are these becoming internal.. that worries me in terms of responsibilities.

Reply:
See ddfccea5b4ebe90bdee03db14a2e0db4df3d8219; I was tired of chasing through internal factory methods since it made it clear as mud what was actually calling these things. The methods were internal, and since they'd always be exposing a specific implementation, there was no real value of the abstraction.

Reply:
ah ok. 👍  on this then.